### ISSUE TITLE
Overriding default SdkClientOption.SCHEDULED_EXECUTOR_SERVICE in AWS SDK for Java 2

### ISSUE THREAD

**Issue by hariharanv01** [aws/aws-sdk-java-v2#1690]
Is there a way I can override the default `SdkClientOption.SCHEDULED_EXECUTOR_SERVICE` setting in AWS SDK for Java 2? 

I want to get rid of all the threadpools that are opaque to the application, and ensure that the threadpools are reused across all non-blocking contexts.

The current behavior in SDK is as below (5 being a magic number)

```java
private ScheduledExecutorService resolveScheduledExecutorService() {
        return Executors.newScheduledThreadPool(5, new ThreadFactoryBuilder()
            .threadNamePrefix("sdk-ScheduledExecutor").build());
}
```

**Comment by debora-ito** [comment]
Seems a reasonable ask, marking as a feature request.

**Comment by Marcus-Rosti** [comment]
I'll second that. When using cached, ephemeral clients, I can see that the scheduled thread pool will at times be leaked when the aws client is evicted by the cache

**Comment by jreznot** [comment]
Especially needed for Spring WebFlux and Project Reactor to reuse boundedElastic thread pool instead of spawning separate threads.

**Comment by RoynniZambrano** [comment]
Hello there, has anyone found a workaround for this?
It would be great to be able to configure it in the same way that the **SdkAdvancedAsyncClientOption.FUTURE_COMPLETION_EXECUTOR** is configured.

**Comment by samarprakash** [comment]
Do you have any updates on this issue? This becomes worse if an application has multiple AWS services in use. For example - if app A uses S3/SQS/SNS/DDB, all clients internally create a separate ScheduledExecutorService with a default core thread pool of 5 and obviously with no allowCoreThreadTimeout and its unnecessary waste of system resources. AWS SDK must provide an option to the client to choose the right behaviour for their application, similar to AsyncConfiguration.

**Comment by debora-ito** [comment]
Feature released in the SDK version `2.20.70` via #4002.
